src/Jackett.Common/Indexers/HDBitsApi.cs

```diff
@@ -89,45 +89,17 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
 
             var categories = MapTorznabCapsToTrackers(query);
 
-            if (categories.Count > 0)
-            {
-                requestData["category"] = new JArray();
-
-                foreach (var cat in categories)
-                {
-                    requestData["category"].Add(new JValue(cat));
-                }
-            }
-
-            if (configData.Codecs.Values.Length > 0)
-            {
-                requestData["codec"] = new JArray();
-
-                foreach (var codec in configData.Codecs.Values)
-                {
-                    requestData["codec"].Add(new JValue(int.Parse(codec)));
-                }
-            }
-
-            if (configData.Mediums.Values.Length > 0)
-            {
-                requestData["medium"] = new JArray();
+            if(categories.Any())
+                requestData.Add("category", JToken.FromObject(categories.Select(int.Parse)));
```

Comment:
In the original code, we did not convert to int before adding. Some trackers have non-numeric categories, so it would be best to keep that same logic.
```suggestion
                requestData.Add("category", JToken.FromObject(categories));
```